src/corehost/cli/hostpolicy.cpp

```diff
@@ -1,279 +1,374 @@
+            {
+                if (fx_curr != fx_begin)
+                    app_context_deps_str += _X(';');
 
-    bool set_app_paths = false;
+                app_context_deps_str += (*fx_curr)->get_deps_file();
+                ++fx_curr;
+            }
 
-    // Runtime options config properties.
-    for (int i = 0; i < g_init.cfg_keys.size(); ++i)
-    {
-        // Provide opt-in compatible behavior by using the switch to set APP_PATHS
-        if (pal::cstrcasecmp(g_init.cfg_keys[i].data(), "Microsoft.NETCore.DotNetHostPolicy.SetAppPaths") == 0)
+            pal::pal_clrstring(app_context_deps_str, &_app_context_deps);
+            pal::pal_clrstring(_resolver.get_lookup_probe_directories(), &_probe_directories);
+
+            if (_resolver.is_framework_dependent())
+            {
+                pal::pal_clrstring(get_root_framework(fx_definitions).get_found_version(), &_clr_library_version);
+            }
+            else
+            {
+                pal::pal_clrstring(_resolver.get_coreclr_library_version(), &_clr_library_version);
+            }
+
+            properties.add(common_property::TrustedPlatformAssemblies, _tpa_paths_cstr.data());
+            properties.add(common_property::NativeDllSearchDirectories, _native_dirs_cstr.data());
+            properties.add(common_property::PlatformResourceRoots, _resources_dirs_cstr.data());
+            properties.add(common_property::AppDomainCompatSwitch, "UseLatestBehaviorWhenTFMNotSpecified");
+            properties.add(common_property::AppContextBaseDirectory, _app_base_cstr.data());
+            properties.add(common_property::AppContextDepsFiles, _app_context_deps.data());
+            properties.add(common_property::FxDepsFile, _fx_deps.data());
+            properties.add(common_property::ProbingDirectories, _probe_directories.data());
+            properties.add(common_property::FxProductVersion, _clr_library_version.data());
+
+            if (!clrjit_path.empty())
+            {
+                pal::pal_clrstring(clrjit_path, &_clrjit_path_cstr);
+                properties.add("JIT_PATH", _clrjit_path_cstr.data());
+            }
+
+            bool set_app_paths = false;
+
+            // Runtime options config properties.
+            for (int i = 0; i < _hostpolicy_init.cfg_keys.size(); ++i)
+            {
+                // Provide opt-in compatible behavior by using the switch to set APP_PATHS
+                if (pal::cstrcasecmp(_hostpolicy_init.cfg_keys[i].data(), "Microsoft.NETCore.DotNetHostPolicy.SetAppPaths") == 0)
+                {
+                    set_app_paths = (pal::cstrcasecmp(_hostpolicy_init.cfg_values[i].data(), "true") == 0);
+                }
+
+                properties.add(_hostpolicy_init.cfg_keys[i].data(), _hostpolicy_init.cfg_values[i].data());
+            }
+
+            // App paths and App NI paths
+            if (set_app_paths)
+            {
+                properties.add("APP_PATHS", _app_base_cstr.data());
```

Comment:
Why not have these properties defined in the enum as well?

Reply:
No real reason other than we optionally set them rather than always set them. The enum was designed to be a place for unified documentation of what should be considered set. In cases where it is a niche case, I left them out. I can add if that is preferred though.

Reply:
I don't feel strongly about this either... feel free to leave it as is... 
I personally would probably add the enums... but it's just a feeling.